tests/scalers/dynatrace/dynatrace_test.go

```diff
@@ -0,0 +1,333 @@
+//go:build e2e
+// +build e2e
+
+package dynatrace_test
+
+import (
+	"encoding/base64"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "dynatrace-test"
+)
+
+var (
+	testNamespace           = fmt.Sprintf("%s-ns", testName)
+	deploymentName          = fmt.Sprintf("%s-deployment", testName)
+	monitoredDeploymentName = fmt.Sprintf("%s-monitored-deployment", testName)
+	serviceName             = fmt.Sprintf("%s-service-%d", testName, GetRandomNumber())
+	scaledObjectName        = fmt.Sprintf("%s-so", testName)
+	secretName              = fmt.Sprintf("%s-secret", testName)
+	triggerAuthName         = fmt.Sprintf("%s-ta", testName)
+	dynatraceHost           = os.Getenv("DYNATRACE_HOST")
+	dynatraceToken          = os.Getenv("DYNATRACE_METRICS_TOKEN")
+	kubernetesClusterName   = "keda-dynatrace-cluster"
+	deploymentReplicas      = 1
+	minReplicaCount         = 0
+	maxReplicaCount         = 2
+)
+
+type templateData struct {
+	TestNamespace           string
+	DeploymentName          string
+	MonitoredDeploymentName string
+	ServiceName             string
+	ScaledObjectName        string
+	TriggerAuthName         string
+	SecretName              string
+	DynatraceToken          string
+	DeploymentReplicas      string
+	DynatraceHost           string
+	KubernetesClusterName   string
+	MinReplicaCount         string
+	MaxReplicaCount         string
+}
+
+const (
+	dynakubeSecretTemplate = `apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+data:
+  apiToken: {{.DynatraceToken}}
+  dataIngestToken: {{.DynatraceToken}}
+`
+
+	dynakubeTemplate = `apiVersion: dynatrace.com/v1beta1
+kind: DynaKube
+metadata:
+name: {{.KubernetesClusterName}}
+namespace: {{.TestNamespace}}
+spec:
+  apiUrl: "{{.DynatraceHost}}/api"
+  networkZone: {{.KubernetesClusterName}}
+  oneAgent:
+    cloudNativeFullStack:
+      args:
+        - --set-host-group={{.KubernetesClusterName}}
+  activeGate:
+    capabilities:
+    - routing
+    - dynatrace-api
+    - metrics-ingest
+    group: {{.KubernetesClusterName}}
+`
+	secretTemplate = `apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.SecretName}}
+  namespace: {{.TestNamespace}}
+data:
+  token: {{.DynatraceToken}}
+`
+
+	triggerAuthenticationTemplate = `apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: {{.TriggerAuthName}}
+  namespace: {{.TestNamespace}}
+spec:
+  secretTargetRef:
+  secretTargetRef:
+  - parameter: token
+    name: {{.SecretName}}
+    key: token
+`
```

Comment:
The key is the secret key
```suggestion
spec:
  secretTargetRef:
  - parameter: token
    name: {{.SecretName}}
    key: apiToken
`
```

Reply:
I had another secret with a different key, didn't realize it was actually overwriting 🤦 I'll keep the one yeah